akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -89,3 +89,11 @@ class DnsExt(system: ExtendedActorSystem) extends IO.Extension {
 
   def getResolver: ActorRef = manager
 }
+
+object IpVersionSelector {
+  def getInetAddress(ipv4: Option[Inet4Address], ipv6: Option[Inet6Address]): Option[InetAddress] =
+    sys.props.get("java.net.preferIPv6Addresses") match {
```

Comment:
Let's not complicate it. The property lookup is probably not so costly compared to the actual dns lookup and with proper caching it shouldn't be done often.